datafusion/common/src/config.rs

```diff
@@ -29,12 +29,16 @@ use std::error::Error;
 use std::fmt::{self, Display};
 use std::str::FromStr;
 
-#[cfg(feature = "parquet")]
+#[cfg(feature = "parquet_encryption")]
```

Comment:
I wonder if we can move all the crypto code into its own module, so we only have to `#cfg` the module rather than so many individual lines 🤔

Reply:
I wasn't able to do this entirely, but I have refactored most of this into methods in encryption.rs, which cleans this up more.